include/nbl/builtin/hlsl/math/quadrature/gauss_legendre/gauss_legendre.hlsl

```diff
@@ -24,26 +25,63 @@ struct GaussLegendreIntegration
 {
     static float_t calculateIntegral(NBL_CONST_REF_ARG(IntegrandFunc) func, float_t start, float_t end)
     {
-        float_t integral = 0.0;
+        float_t integral = _static_cast<float_t>(0ull);
         for (uint32_t i = 0u; i < Order; ++i)
         {
-            const float_t xi = GaussLegendreValues<Order, float_t>::xi(i) * ((end - start) / 2.0) + ((end + start) / 2.0);
-            integral += GaussLegendreValues<Order, float_t>::wi(i) * func(xi);
+            const float_t xi = GaussLegendreValues<Order, float_t>::xi(i) * ((end - start) / 2.0f) + ((end + start) / 2.0f);
+            integral = integral + GaussLegendreValues<Order, float_t>::wi(i) * func(xi);
         }
+
         return ((end - start) / 2.0) * integral;
     }
 };
 
 #define float_t float32_t
+#define float_t_namespace impl_float32_t
 #define TYPED_NUMBER(N) NBL_CONCATENATE(N, f) // to add f after floating point numbers and avoid casting warnings and emitting ShaderFloat64 Caps
 #include <nbl/builtin/hlsl/math/quadrature/gauss_legendre/impl.hlsl>
 #undef TYPED_NUMBER
+#undef float_t_namespace
 #undef float_t
 
 #define float_t float64_t
+#define float_t_namespace impl_float64_t
 #define TYPED_NUMBER(N) N
 #include <nbl/builtin/hlsl/math/quadrature/gauss_legendre/impl.hlsl>
 #undef TYPED_NUMBER
+#undef float_t_namespace
+#undef float_t
+
+#define float_t emulated_float64_t<true, true>
+#define float_t_namespace impl_emulated_float64_t_true_true
+#define TYPED_NUMBER(N) emulated_float64_t<true, true>::create(N)
+#include <nbl/builtin/hlsl/math/quadrature/gauss_legendre/impl.hlsl>
+#undef TYPED_NUMBER
+#undef float_t_namespace
+#undef float_t
+
+#define float_t emulated_float64_t<false, false>
+#define float_t_namespace impl_emulated_float64_t_false_false
+#define TYPED_NUMBER(N) emulated_float64_t<false, false>::create(N)
+#include <nbl/builtin/hlsl/math/quadrature/gauss_legendre/impl.hlsl>
+#undef TYPED_NUMBER
+#undef float_t_namespace
+#undef float_t
+
+#define float_t emulated_float64_t<false, true>
+#define float_t_namespace impl_emulated_float64_t_false_true
+#define TYPED_NUMBER(N) emulated_float64_t<false, true>::create(N)
+#include <nbl/builtin/hlsl/math/quadrature/gauss_legendre/impl.hlsl>
+#undef TYPED_NUMBER
+#undef float_t_namespace
+#undef float_t
+
+#define float_t emulated_float64_t<true, false>
+#define float_t_namespace impl_emulated_float64_t_true_false
+#define TYPED_NUMBER(N) emulated_float64_t<true, false>::create(N)
+#include <nbl/builtin/hlsl/math/quadrature/gauss_legendre/impl.hlsl>
+#undef TYPED_NUMBER
+#undef float_t_namespace
 #undef float_t
```

Comment:
I think if you declared the float64 array once, but as emulated floats, then you could make the regular native floats just bit_cast read the emualted ones